Sources/joyfill-components-swift/Components/ComponentViewController/JSONComponents/ComponentsViewController.swift

```diff
@@ -0,0 +1,252 @@
+import Foundation
+import UIKit
+
+public class ComponentsViewController: UIView {
+    
+    public var componentView = UIView()
+    public var componentTableView = UITableView()
+    
+    var cellView = [UIView]()
+    var cellHeight = [CGFloat]()
+    
+    // MARK: Initializer
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        setupUI()
+    }
+    
+    public required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        setupUI()
+    }
+    
+    public init() {
+        super.init(frame: .zero)
+        setupUI()
+    }
+    
+    func setupUI() {
+        // Deinitialize arrays to protect memory leakage
+        yCoordinates = []
+        xCoordinates = []
+        graphLabelData = []
+        pickedImg.removeAll()
+        _ = JoyfillApi2.loadFromJSON()
+        
+        // SubViews
+        addSubview(componentView)
+        componentView.addSubview(componentTableView)
+        componentView.translatesAutoresizingMaskIntoConstraints = false
+        componentTableView.translatesAutoresizingMaskIntoConstraints = false
+        
+        // Constraint to arrange subviews acc. to components
+        NSLayoutConstraint.activate([
+            // ComponentView Constraint
+            componentView.topAnchor.constraint(equalTo: safeAreaLayoutGuide.topAnchor, constant: 20),
+            componentView.leadingAnchor.constraint(equalTo: leadingAnchor, constant: 10),
+            componentView.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -10),
+            componentView.bottomAnchor.constraint(equalTo: safeAreaLayoutGuide.bottomAnchor, constant: -20),
+            
+            // TableView Constraint
+            componentTableView.topAnchor.constraint(equalTo: componentView.topAnchor, constant: 9),
+            componentTableView.leadingAnchor.constraint(equalTo: componentView.leadingAnchor, constant: 0),
+            componentTableView.trailingAnchor.constraint(equalTo: componentView.trailingAnchor, constant: 0),
+            componentTableView.bottomAnchor.constraint(equalTo: componentView.bottomAnchor, constant: -10)
+        ])
+        
+        // Set cornerRadius and shadow to view.
+        backgroundColor = .white
+        componentView.backgroundColor = .white
+        componentView.layer.cornerRadius = 15
+        componentView.layer.shadowOpacity = 0.3
```

Comment:
What does the componentView reference

Reply:
It's the view on which all the components will render